src/sched.rs

```diff
@@ -215,6 +215,112 @@ mod sched_linux_like {
 
         Errno::result(res).map(drop)
     }
+
+    libc_bitflags! {
+        pub struct SchedFlags: c_int {
+            #[cfg(target_os = "android")]
+            SCHED_NORMAL;
+            #[cfg(target_os = "linux")]
+            SCHED_OTHER;
+            SCHED_FIFO;
+            SCHED_RR;
+            SCHED_BATCH;
+            SCHED_IDLE;
+            #[cfg(target_os = "android")]
+            SCHED_DEADLINE;
+            #[cfg(target_os = "linux")]
+            SCHED_RESET_ON_FORK;
+        }
+    }
+
+    #[repr(C)]
+    #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+    pub struct SchedParam(libc::sched_param);
+
+    impl SchedParam {
+        pub fn new(priority: i32) -> Self {
+            let mut sched_param: libc::sched_param = unsafe { mem::zeroed() };
+            sched_param.sched_priority = priority;
+            SchedParam(sched_param)
+        }
+
+        pub fn priority(&self) -> i32 {
+            self.0.sched_priority
+        }
+    }
+
+    impl Default for SchedParam {
```

Comment:
You should be able to omit this part, and just add `#[derive(Default)]` to the struct.

Reply:
`libc::unix::linux_like::sched_param` does not implement `std::default::Default`, so this fails with a compiler error.